### PR TITLE
Feature/boost clean up

### PR DESCRIPTION
**Pull request by Baklap4**
### What does this PR do?
Cleaned up some boost ptrs and unused usings.
Cleans up all occurences of `boost::ptr_vector` and `boost::ptr_map` replaced by c++11 equivalents. Also gets rid of the `cachemap.hh` which was just a plain wrapper around `boost::ptr_map`

Also cherry-picked: 834e21efab0140194196240ca47c95bd48a97949 (std::function replacement) and fixed the merge-conflict in `video_driver.hh` 

### Closes Issue(s)
Does not close but helps #39 

### Motivation
Cleaning up should be done sometime.

### CODE REVIEW

File: game/opengl_text.cc [hunk 1]
```diff
@@ -263,15 +263,16 @@ void SvgTxtTheme::draw(std::vector<TZoomText>& _text) {
 		for (auto& zt: _text) {
 			if (zt.wordStart == true) { zt.addSpace(); } // If it's a different word, add an extra space to account for lyrics zooming in.
 			m_text.text = zt.string;
-			m_opengl_text.push_back(new OpenGLText(m_text, m_factor));
+			auto openGlPtr = std::unique_ptr<OpenGLText>(new OpenGLText(m_text, m_factor));
```
Comment (Lord-Kamina) [inline game/opengl_text.cc]: Please use make_unique instead of new.

File: game/screen_practice.cc [hunk 2]
```diff
@@ -15,7 +15,8 @@ void ScreenPractice::enter() {
 	m_audio.playMusic(findFile("practice.ogg"));
 	// draw vu meters
 	for (unsigned int i = 0, mics = m_audio.analyzers().size(); i < mics; ++i) {
-		m_vumeters.push_back(new ProgressBar(findFile("vumeter_bg.svg"), findFile("vumeter_fg.svg"), ProgressBar::VERTICAL, 0.136, 0.023));
+		auto progressBarPtr = std::unique_ptr<ProgressBar>(new ProgressBar(findFile("vumeter_bg.svg"), findFile("vumeter_fg.svg"), ProgressBar::VERTICAL, 0.136, 0.023));
```
Comment (Lord-Kamina) [inline game/screen_practice.cc]: See above.

File: game/screen_sing.cc [hunk 3]
```diff
@@ -123,7 +123,7 @@ void ScreenSing::setupVocals() {
 			shownTracks.insert(vocal);
 		}
 		//if (shownTracks.size() > 2) throw std::runtime_error("Too many tracks chosen. Only two vocal tracks can be used simultaneously.");
-		for (auto const& trk: shownTracks) m_layout_singer.push_back(new LayoutSinger(*trk, m_database, theme));
+		for (auto const& trk: shownTracks) m_layout_singer.push_back(std::move(std::unique_ptr<LayoutSinger>(new LayoutSinger(*trk, m_database, theme))));
```
Comment (Lord-Kamina) [inline game/screen_sing.cc]: Same.

**Comment by Lord-Kamina** [comment]
I think you don't need to use auto; `auto openGlPtr = std::unique_ptr<OpenGLText>(std::make_unique<OpenGLText>(m_text, m_factor));` is probably better as `std::unique_ptr<OpenGLText> openGlPtr = std::make_unique<OpenGLText>(m_text, m_factor);` for example.

Also, please use `using` instead of `typedef`.  AFAIK they're pretty much equivalent but the newer syntax is more readable IMO.

**Comment by Baklap4** [comment]
Cleans up all occurences of `boost::ptr_vector` and `boost::ptr_map` replaced by c++11 equivalents. Also gets rid of the `cachemap.hh` which was just a plain wrapper around `boost::ptr_map` Jeej clean code!

**Comment by Baklap4** [comment]
Fixed @Lord-Kamina

**Comment by Lord-Kamina** [comment]
Looks good. I still think we need to change `typedef` to `using` but seeing my cherry-picked commit I see I made the same mistake in my PR so maybe we can make another one later to  homogenize all instances.

**Pull request merged**